test/integration/02-user-wallet/02-send-onchain.spec.ts

```diff
@@ -195,147 +241,237 @@ describe("UserWallet - onChainPay", () => {
+  const initialRecipientBalance = await getBalanceHelper(recipientWalletId)
 
-    let pendingTxHash
+  const address = await Wallets.createOnChainAddress(recipientWalletId)
+  if (address instanceof Error) return address
 
-    {
-      const txResult = await Wallets.getTransactionsForWalletId({
-        walletId: walletIdE,
-      })
-      if (txResult.error instanceof Error || txResult.result === null) {
-        throw txResult.error
-      }
-      const pendingTxs = txResult.result.filter(
-        ({ status }) => status === TxStatus.Pending,
-      )
-      expect(pendingTxs.length).toBe(1)
-      const pendingTx = pendingTxs[0]
-      expect(pendingTx.settlementAmount).toBe(-initialBalanceUserE)
+  const paid = await Wallets.payOnChainByWalletId({
+    senderAccount: senderAccount,
+    senderWalletId: senderWalletId,
+    address,
+    amount: senderAmount,
+    targetConfirmations,
+    memo,
+    sendAll: false,
+  })
+  if (paid instanceof Error) return paid
 
-      pendingTxHash = pendingTx.id as OnChainTxHash
+  // Check balances for both wallets
+  // ===
+  const finalSenderBalance = await getBalanceHelper(senderWalletId)
+  const finalRecipient = await getBalanceHelper(recipientWalletId)
 
-      const interimBalance = await getBalanceHelper(walletIdE)
-      expect(interimBalance).toBe(0)
-      await checkIsBalanced()
-    }
+  expect(paid).toBe(PaymentSendStatus.Success)
+  expect(finalSenderBalance).toBe(initialSenderBalance - senderAmount)
+  expect(finalRecipient).toBe(initialRecipientBalance + recipientAmount)
 
-    // const subSpend = subscribeToChainSpend({ lnd: lndonchain, bech32_address: address, min_height: 1 })
+  // Check txn details for sent wallet
+  // ===
+  const { result: txsSender, error } = await Wallets.getTransactionsForWalletId({
+    walletId: senderWalletId,
+  })
+  if (error instanceof Error || txsSender === null) {
+    return error
+  }
+  const pendingTxsSender = txsSender.filter(({ status }) => status === TxStatus.Pending)
+  expect(pendingTxsSender.length).toBe(0)
+
+  const settledTxsSender = txsSender.filter(
+    ({ status, initiationVia, settlementVia, memo: txMemo }) =>
+      status === TxStatus.Success &&
+      initiationVia.type === PaymentInitiationMethod.OnChain &&
+      settlementVia.type === SettlementMethod.IntraLedger &&
+      txMemo === memo,
+  )
+  expect(settledTxsSender.length).toBe(1)
+  const senderSettledTx = settledTxsSender[0] as WalletTransaction
+
+  expect(senderSettledTx.settlementFee).toBe(0)
+  expect(senderSettledTx.settlementAmount).toBe(-senderAmount)
+  expect(senderSettledTx.displayCurrencyPerSettlementCurrencyUnit).toBeGreaterThan(0)
+
+  // Check txn details for received wallet
+  // ===
+  const { result: txsRecipient, error: errorUserA } =
+    await Wallets.getTransactionsForWalletId({
+      walletId: recipientWalletId,
+    })
+  if (errorUserA instanceof Error || txsRecipient === null) {
```

Comment:
if `txsRecipient === null` it won't be catch as an error

Reply:
This is more for the type catcher here. We've done this in a number of places in the tests. With the way the `Partial` return from `getTransactionsForWalletId` works there can never be a `null` txsRecipient value without there being an `errorUserA`.

We could probably clean up how we call that method in follow-up PR if we want to make that clearer.